File: MPChartExample/src/com/xxmassdeveloper/mpchartexample/notimportant/MyAdapter.java
package com.xxmassdeveloper.mpchartexample.notimportant;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.Typeface;
import androidx.annotation.NonNull;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.xxmassdeveloper.mpchartexample.R;

import java.util.List;

/**
 * Created by Philipp Jahoda on 07/12/15.
 */
public class MyAdapter extends ArrayAdapter<ContentItem> {

    private final Typeface mTypeFaceLight;
    private final Typeface mTypeFaceRegular;

    public MyAdapter(Context context, List<ContentItem> objects) {
        super(context, 0, objects);

        mTypeFaceLight = Typeface.createFromAsset(context.getAssets(), "OpenSans-Light.ttf");
        mTypeFaceRegular = Typeface.createFromAsset(context.getAssets(), "OpenSans-Regular.ttf");
    }

    @SuppressLint("InflateParams")
    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {

        ContentItem c = getItem(position);

        ViewHolder holder;

        holder = new ViewHolder();

        if (c != null && c.isSection) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.list_item_section, null);
        } else {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.list_item, null);
        }

        holder.tvName = convertView.findViewById(R.id.tvName);
        holder.tvDesc = convertView.findViewById(R.id.tvDesc);

        convertView.setTag(holder);

        if (c != null && c.isSection)
            holder.tvName.setTypeface(mTypeFaceRegular);
        else
            holder.tvName.setTypeface(mTypeFaceLight);
        holder.tvDesc.setTypeface(mTypeFaceLight);

        holder.tvName.setText(c != null ? c.name : null);
        holder.tvDesc.setText(c != null ? c.desc : null);

        return convertView;
    }

    private class ViewHolder {

        TextView tvName, tvDesc;
    }
}

File: MPChartExample/src/main/java/com/xxmassdeveloper/mpchartexample/notimportant/MyAdapter.kt
package com.xxmassdeveloper.mpchartexample.notimportant

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.xxmassdeveloper.mpchartexample.R

/**
 * Created by Philipp Jahoda on 07/12/15.
 */
internal class MyAdapter(context: Context, objects: List<ContentItem>?) :
    ArrayAdapter<ContentItem?>(context, 0, objects!!) {
    private val mTypeFaceLight: Typeface
    private val mTypeFaceRegular: Typeface
    @SuppressLint("InflateParams")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val c = getItem(position)
        val holder: ViewHolder
        holder = ViewHolder()
        convertView = if (c != null && c.isSection) {
            LayoutInflater.from(context).inflate(R.layout.list_item_section, null)
        } else {
            LayoutInflater.from(context).inflate(R.layout.list_item, null)
        }
        holder.tvName = convertView.findViewById(R.id.tvName)
        holder.tvDesc = convertView.findViewById(R.id.tvDesc)
        convertView.tag = holder
        if (c != null && c.isSection) holder?.tvName?.setTypeface(mTypeFaceRegular) else holder?.tvName?.setTypeface(
            mTypeFaceLight
        )
        holder?.tvDesc?.setTypeface(mTypeFaceLight)
        holder?.tvName?.text = c?.name
        holder?.tvDesc?.text = c?.desc
        return convertView
    }

    private inner class ViewHolder {
        var tvName: TextView? = null
        var tvDesc: TextView? = null
    }

    init {
        mTypeFaceLight = Typeface.createFromAsset(context.assets, "OpenSans-Light.ttf")
        mTypeFaceRegular = Typeface.createFromAsset(context.assets, "OpenSans-Regular.ttf")
    }
}